Documentation/design-docs/multilevel-sharedfx-lookup.md

```diff
@@ -105,6 +105,67 @@ and Microsoft.AspNetCore.App's runtimeconfig.json would contain:
 ```
 and Microsoft.NETCore.App would not have a runtimeconfig.json because it doesn't have any framework dependency or need to change settings.
 
+### Proposed changes for 3.0 (specifying multiple frameworks)
+The 2.1 release added support for a chain of frameworks, where each framework can have one dependent framework. However, with the advent of frameworks for WPF and WinForms it becomes necessary for an application to be able to reference more than one dependent framework.
+
+The runtimeconfig.json will have a new `frameworks` array section that allows more than one framework to be specified:
+```javascript
+"runtimeOptions": {
+	"frameworks": [
+			{
+					"name": "Microsoft.AspNetCore.All",
+					"version": "3.0.0"
+			},
+			{
+					"name": "Microsoft.Forms",
+					"version": "3.0.0"
+					"rollForwardOnNoCandidateFx" : 1
+					"applyPatches" : true
+			}
+	]
+}
+```
+
+If an entry also exists in the `framework` section, it is treated as the first element in the `frameworks` array. The `framework` section is no longer required but is supported for backwards compatibility.
+
+The `applyPatches` and `rollForwardOnNoCandidateFx` can be specified individually for each framework. These per-framework settings override any corresponding values that may be present in the `runtimeOptions` section.
+
+By allowing more than one framework reference, we may encounter issues with multiple references to the same framework but with different versions or with different roll-forward settings. The rules to reconcile that include:
```

Comment:
This language is very vague.. given the importance of the resolution logic, we should be very clear of how it works. For example the way it's written now suggests that there might be additional rules which are not described here.

Reply:
Sure. I could describe the algorithm as I see it, which is how it exists in the current prototype. 

One non-intuitive item is that we may have to "start over" (re-resolve the frameworks) if we encounter a higher version or more constrained roll-forward setting for a framework that we already resolved.

Reply:
I think it might be wroth doing both in a way. Describe the high-level behavior - which is basically what you already have here.
And then have a proposal of how to implement that algorithmically - which would point out the problems with the restarts. But maybe that can wait for the actual code-review? Not sure.